app/controllers/names_controller.rb

```diff
@@ -504,11 +512,12 @@ def update_correct_spelling
   end
 
   def set_correct_spelling
-    correct_name = Name.find_names_filling_in_authors(@correct_spelling).first
+    correct_name = Name.find_names_filling_in_authors(@user,
+                                                      @correct_spelling).first
     raise(:runtime_form_names_misspelling_bad.t) unless correct_name
     raise(:runtime_form_names_misspelling_same.t) if correct_name.id == @name.id
 
-    @name.mark_misspelled(correct_name)
+    @name.mark_misspelled(@user, correct_name)
```

Comment:
To me it seems a little unintuitive to make `user` now the first arg of a name method. Thankfully I don't deal with these much, but if i were calling `@name.mark_misspelled(*args)` I would expect `(correct_name, @user)`

Reply:
I guess you're doing it everywhere, and trying to foreground the fact that this method depends on the current user.

Reply:
It's more about having a simple way to get to the end goal of getting rid of User.current.  See above comment about functions which slurp up remaining arguments.